cpp/tests/reductions/reduction_tests.cpp

```diff
@@ -2501,15 +2479,13 @@ TYPED_TEST(DictionaryReductionTest, DISABLED_VarStd)
   std::vector<bool> validity({true, true, false, true, true, true, false, true});
   cudf::test::dictionary_column_wrapper<T> col_nulls(v.begin(), v.end(), validity.begin());
 
-  cudf::size_type const valid_count = std::count(validity.begin(), validity.end(), true);
-
-  double var_nulls = calc_var(replace_nulls(v, validity, T{0}), valid_count, ddof);
+  double var_nulls = calc_var(v, ddof, validity);
   double std_nulls = std::sqrt(var_nulls);
 
-  EXPECT_EQ(this->template reduction_test<double>(col_nulls, *var_agg, output_type).first,
-            var_nulls);
-  EXPECT_EQ(this->template reduction_test<double>(col_nulls, *std_agg, output_type).first,
-            std_nulls);
+  EXPECT_DOUBLE_EQ(this->template reduction_test<double>(col_nulls, *var_agg, output_type).first,
+                   var_nulls);
+  EXPECT_DOUBLE_EQ(this->template reduction_test<double>(col_nulls, *std_agg, output_type).first,
+                   std_nulls);
```

Comment:
I think we were just lucky before that the result we obtained was exactly the same. Now the host-side calculation does the two-pass algorithm:

$$
\frac{1}{n - \text{ddof}}\sum_{i=1}^{n} (x_i - \bar{x})^2
$$

So this has some ulp differences to the one-pass algorithm on the device side. `EXPECT_DOUBLE_EQ` allows for up to 4ULP difference.

Reply:
I had no idea that Github markdown supported this sort of notation.